ssl/statem/statem_lib.c

```diff
@@ -1871,3 +1875,37 @@ int check_in_list(SSL *s, unsigned int group_id, const unsigned char *groups,
     return i < num_groups;
 }
 #endif
+
+/* Replace ClientHello1 in the transcript hash with a synthetic message */
+int create_synthetic_message_hash(SSL *s)
+{
+    unsigned char hashval[EVP_MAX_MD_SIZE];
+    size_t hashlen = 0;
+    unsigned char msghdr[SSL3_HM_HEADER_LENGTH] = {
```

Comment:
if this doesn't exactly match the header length we won't necesarily know and there will be garbage values.  I prefer memset/assign or this:
```
static unsigned char nilmsghdr[SSL3_HM_HEADER_LENGTH] = { SSL3_MT_MESSAGE_HASH };
msghdr = nilmsghdr;
```
(You get a guarantee of zero-fill for static, not for locals.)